src/basic/fd-util.c

```diff
@@ -208,7 +208,7 @@ static int get_max_fd(void) {
         return (int) (m - 1);
 }
 
-int close_all_fds(int except[], size_t n_except) {
+int close_all_fds_full(int except[], size_t n_except, bool allow_alloc) {
```

Comment:
this is a bit weird. this should be an inner function, and a wrapper, as mentioned